eZ/Bundle/EzPublishCoreBundle/Features/Context/FieldTypeContext.php

```diff
@@ -0,0 +1,407 @@
+        $this->fieldConstructionObject['objectState'] = self::FIELD_TYPE_CREATED;
+    }
+
+    /**
+     * Adds a validator to the stored field.
+     *
+     * @param string $fieldType Type of the field
+     * @param string $value Value of the constraint
+     * @param string $constraint Constraint name
+     */
+    public function addValueConstraint($fieldType, $value, $constraint)
+    {
+        $validatorName = $this->getFieldValidator($fieldType);
+        $validatorParent = $validatorName . 'Validator';
+        if ($this->fieldConstructionObject['fieldType']->validatorConfiguration == null) {
+            $this->fieldConstructionObject['fieldType']->validatorConfiguration = array(
+                $validatorParent => array(),
+            );
+        }
+        $value = is_numeric($value) ? $value + 0 : $value;
+
+        $this->fieldConstructionObject['fieldType']
+            ->validatorConfiguration[$validatorParent][$constraint . $validatorName] = $value;
+    }
+
+    /**
+     * Creates a content and publishes it.
+     *
+     * @param string $field Name of the field
+     * @param mixed $value Value of the field
+     */
+    public function createContent($field, $value)
+    {
+        $this->setFieldContentState(self::CONTENT_PUBLISHED, $field, $value);
+    }
+
+    /**
+     * Executes necessary operations to guarantee a given state, recursive
+     * function that calls it self to make sure prerequisites are met.
+     *
+     * @param int $stateFlag Desired state, only predefined constants accepted
+     * @param string $field Name of the field, optional
+     * @param mixed $value Value of the field, optional
+     */
+    public function setFieldContentState($stateFlag, $field = null, $value = null)
+    {
+        if ($stateFlag <= $this->fieldConstructionObject['objectState']
+            || $stateFlag < self::FIELD_TYPE_NOT_CREATED
+        ) {
+            return;
+        }
+
+        // recursively set previous states if necessary
+        $this->setFieldContentState($stateFlag - 1, $field, $value);
+
+        switch ($stateFlag) {
+            case self::FIELD_TYPE_NOT_CREATED:
+                throw new \Exception('A field type must be declared before anything else');
+                break;
+            case self::CONTENT_TYPE_CREATED:
+                $this->createContentType();
+                break;
+            case self::FIELD_TYPE_ASSOCIATED:
+                $this->associateFieldToContentType();
+                break;
+            case self::CONTENT_TYPE_PUBLISHED:
+                $this->publishContentType();
+                break;
+            case self::CONTENT_PUBLISHED:
+                $this->publishContent($field, $value);
+                break;
+        }
+    }
+
+    public function getFieldContentState()
+    {
+        return $this->fieldConstructionObject['objectState'];
+    }
+
+    /**
+     * Publishes the content.
```

Comment:
Based on the description alone: what "content" does it publish ? As far as I can tell, it creates and published a Content Item based on the internally configured contentType, and it may set a given field to a given value if they're provided.